src/main/java/org/opensearch/security/resources/rest/ResourceAccessRestAction.java

```diff
@@ -0,0 +1,154 @@
+                new Route(POST, "/revoke"),
+                new Route(POST, "/share"),
+                new Route(POST, "/verify_access")
+            ),
+            PLUGIN_RESOURCE_ROUTE_PREFIX
+        );
+    }
+
+    @Override
+    public String getName() {
+        return "resource_api_action";
+    }
+
+    @Override
+    protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient client) throws IOException {
+        consumeParams(request); // early consume params to avoid 400s
+
+        Map<String, Object> source = new HashMap<>();
+        if (request.hasContent()) {
+            try (XContentParser parser = request.contentParser()) {
+                source = parser.map();
+            }
+        }
+
+        String path = request.path().split(PLUGIN_RESOURCE_ROUTE_PREFIX)[1].split("/")[1];
+        switch (path) {
+            case "list" -> source.put("operation", LIST);
+            case "revoke" -> source.put("operation", REVOKE);
+            case "share" -> source.put("operation", SHARE);
+            case "verify_access" -> source.put("operation", VERIFY);
+            default -> {
+                return channel -> badRequest(channel, "Unknown route: " + path);
+            }
+        }
+
+        ResourceAccessRequest resourceAccessRequest = ResourceAccessRequest.from(source, request.params());
+        return channel -> {
+            client.executeLocally(ResourceAccessAction.INSTANCE, resourceAccessRequest, new ActionListener<>() {
+
+                @Override
+                public void onResponse(ResourceAccessResponse response) {
+                    try {
+                        sendResponse(channel, response);
+                    } catch (IOException e) {
+                        handleError(channel, e);
+                    }
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    handleError(channel, e);
+                }
+
+            });
+        };
+    }
+
+    /**
+     * Consume params early to avoid 400s.
+     *
+     * @param request from which the params must be consumed
+     */
+    private void consumeParams(RestRequest request) {
+        request.param("resource_index", "");
+    }
+
+    /**
+     * Send the appropriate response to the channel.
+     * @param channel the channel to send the response to
+     * @param response the response to send
+     * @throws IOException if an I/O error occurs
+     */
+    private void sendResponse(RestChannel channel, ResourceAccessResponse response) throws IOException {
+        ok(channel, response::toXContent);
+    }
+
+    /**
+     * Handle errors that occur during request processing.
+     * @param channel the channel to send the error response to
+     * @param e the exception that caused the error
+     */
+    private void handleError(RestChannel channel, Exception e) {
+        String message = e.getMessage();
+        LOGGER.error(message, e);
```

Comment:
Non 500 errors are expected states and should not be logged with an `error` level. Only actual unexpected states should be logged on error levels.

Reply:
okay.